service/operator/service.go

```diff
@@ -163,116 +159,331 @@ func (s *Service) addFunc(obj interface{}) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	err := s.addFuncError(obj)
+	err := ProcessCreate(obj, s)
```

Comment:
All changes in the service go towards implementing the `Operator` interface. The PR is more or less ugly because I did a roundhousekick to have my own creative playground without the need to think about how to structure changes. I did not know what and how to change in the beginning. This is also why the commits are more or less useless, especially the first ones. I only used them to safe my progress remotely. For reviews you may want to look at this file as a whole, since the diff may also not be that helpful.